File: vespajlib/src/main/java/com/yahoo/slime/NixValue.java
// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
package com.yahoo.slime;

/**
 * @author havardpe
 */
final class NixValue extends Value {

    private static final NixValue invalidNix = new NixValue();
    private static final NixValue validNix = new NixValue();
    private NixValue() {}
    public Type type() { return Type.NIX; }
    public void accept(Visitor v) {
        if (valid()) {
            v.visitNix();
        } else {
            v.visitInvalid();
        }
    }
    public static NixValue invalid() { return invalidNix; }
    public static NixValue instance() { return validNix; }

}

File: client/go/internal/slime/nix_value.go
// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
// Author: arnej

package slime

type NixValue struct {
	valueBase
	valid bool
}

func (n *NixValue) Valid() bool {
	return n.valid
}

func (n *NixValue) Type() Type {
	return NIX
}

// actually constants:

var ValidNix *NixValue = &NixValue{valid: true}
var InvalidNix *NixValue = &NixValue{valid: false}
